### PR TITLE
populate resolv.conf with dnsname responses when in usernamespace

### PR DESCRIPTION
**Pull request by baude**
when using usernamespace, dnsname respondes from cni were not making it into the containers /etc/resolv.conf because of a timing issue.  this corrects that behavior.

Fixes: #5256

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

File: libpod/container_internal.go [hunk 1]
```diff
@@ -927,7 +928,37 @@ func (c *Container) completeNetworkSetup() error {
...
+	for _, cni := range state.NetworkStatus {
+		if cni.DNS.Nameservers != nil {
+			for _, server := range cni.DNS.Nameservers {
+				outResolvConf = append(outResolvConf, fmt.Sprintf("nameserver %s", server))
+			}
+		}
+	}
+	// check if we have a bindmount for resolv.conf
+	resolvBindMount := state.BindMounts["/etc/resolv.conf"]
+	if len(outResolvConf) < 1 || resolvBindMount != "" || len(c.config.NetNsCtr) > 0 {
```
Comment (mheon) [inline libpod/container_internal.go]: I think `resolvBindMount` is flipped - we only want to return nil if it is ""

File: libpod/container_internal.go [hunk 2]
```diff
@@ -927,7 +928,37 @@ func (c *Container) completeNetworkSetup() error {
...
+	}
+	for _, line := range strings.Split(string(b), "\n") {
+		// only keep things that dont start with nameserver from the old
+		// resolv.conf file
+		if !strings.HasPrefix(line, "nameserver") {
+			outResolvConf = append([]string{line}, outResolvConf...)
+		}
+	}
+	// write and return
+	return ioutil.WriteFile(resolvBindMount, []byte(strings.Join(outResolvConf, "\n")), 0644)
```
Comment (mheon) [inline libpod/container_internal.go]: This overwrites, and doesn't delete, right?

Comment (baude) [reply libpod/container_internal.go]: i believe that is correct.

**Comment by rhatdan** [comment]
LGTM
@mheon @giuseppe @TomSweeneyRedHat @jwhonce @QiWang19 PTAL

**Pull request merged**